eval.py

```diff
@@ -2,47 +2,108 @@
+        idx = parquet_file["clip_id"].tolist().index(selected_id)
+        train_x.append(parquet_file["mswc_embedding_vector"][idx])
+        train_y.append(training_classes.index("nontarget"))
+
+    for label in selected_data["targets"].keys():
+        if not label in training_classes:
+            raise ValueError('Provided label doesnt match classes in eval set')
+        for selected_id in selected_data["targets"][label]:
+            if not selected_id in allowed_training_embeddings['targets'][label]:
+                print("invaid embedding ID: ", selected_id)
+                raise ValueError('Selected Embedding ID not present in allowed training data')
+            if not label == Path(selected_id).parts[label_idx]:
+                print("invaid label for embedding ID: ", selected_id)
+                raise ValueError('Provided label doesnt match label in ID')
+
+            parquet_file = pd.read_parquet(train_dataset / (label + ".parquet"))
+            idx = parquet_file["clip_id"].tolist().index(selected_id)
+            train_x.append(parquet_file["mswc_embedding_vector"][idx])
+            train_y.append(training_classes.index(label))
+    return train_x, train_y
+
+def create_eval_sets(eval_data, eval_dataset, training_classes):
+    eval_x = []
+    eval_y = []
+    for id in eval_data["nontargets"]:
+        label = Path(id).parts[label_idx]
+        parquet_file = pd.read_parquet(eval_dataset / (label + ".parquet"))
+        idx = parquet_file["clip_id"].tolist().index(id)
+        eval_x.append(parquet_file["mswc_embedding_vector"][idx])
+        eval_y.append(training_classes.index("nontarget"))
+    for label in eval_data["targets"].keys():
+        parquet_file = pd.read_parquet(eval_dataset / (label + ".parquet"))
+        for id in eval_data["targets"][label]:
+            idx = parquet_file["clip_id"].tolist().index(id)
+            eval_x.append(parquet_file["mswc_embedding_vector"][idx])
+            eval_y.append(training_classes.index(label))
+    return eval_x, eval_y
+
+def main(   
+            eval_embeddings_dir="embeddings/en", #embeddings dir point to the same parquet file for testing and online eval
+            train_embeddings_dir="embeddings/en", 
+            allowed_training_set="allowed_training_set.yaml", 
+            eval_file="eval.yaml", 
+            train_file="workdir/train.yaml", 
+            config_file="dataperf_speech_config.yaml"):
+
+    with open(config_file, "r") as fh:
+        config = yaml.safe_load(fh)
+    train_set_size_limit = config["train_set_size_limit"]
+    #TODO add more config parameters
+    
+    with open(allowed_training_set, "r") as fh:
+        allowed_training_embeddings = yaml.safe_load(fh) #dict {"targets": {"dog":[list]}, "nontargets": [list]}
+    training_classes = list(allowed_training_embeddings["targets"].keys())#specifies the label indicies for training 
+    training_classes.append("nontarget")
```

Comment:
this is just a stylistic choice but what do you think about prepending the list so that `nontarget` is always id `0`?

Reply:
Yeah I agree that feels right